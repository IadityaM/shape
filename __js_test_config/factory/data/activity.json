{"data":{"id":"1365201","type":"activities","attributes":{"action":null,"created_at":"2019-12-16T23:14:35.973Z","content":null,"actor_id":22548,"target_type":"Collection","target_id":762911,"source_name":""},"relationships":{"actor":{"meta":{"included":false}},"subject_users":{"meta":{"included":false}},"subject_groups":{"meta":{"included":false}},"target":{"meta":{"included":false}}}}}