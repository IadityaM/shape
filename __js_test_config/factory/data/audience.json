{"data":{"id":"104","type":"audiences","attributes":{"name":"Natural Armor","global_default":null,"age_list":[],"children_age_list":[],"country_list":[],"education_level_list":[],"gender_list":[],"adopter_type_list":[],"interest_list":[],"publication_list":[],"min_price_per_response":4.0,"order":null,"global":true}}}