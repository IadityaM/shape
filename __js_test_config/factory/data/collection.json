{"data":{"id":"762903","type":"collections","attributes":{"class_type":"Collection","name":"Extended","created_at":"2019-12-16T23:14:33.643Z","updated_at":"2019-12-16T23:14:33.643Z","master_template":false,"template_id":null,"submission_box_type":null,"submission_box_id":null,"submission_template_id":null,"test_status":null,"collection_to_test_id":null,"hide_submissions":false,"submissions_enabled":true,"anyone_can_view":false,"anyone_can_join":false,"cover_type":"cover_type_default","archived":false,"unresolved_count":0,"last_unresolved_comment_id":null,"test_show_media":true,"idea_id":null,"search_term":null,"organization_id":"5352","joinable_group_id":null,"default_group_id":null,"system_required":false,"tag_list":[],"inherited_tag_list":[],"cover":{"text":"","hardcoded_subtitle":"","subtitle_hidden":false},"test_scores":{},"breadcrumb":[{"type":"collections","id":"762903","name":"Extended","can_edit":false}],"processing_status":null,"collection_card_count":0,"can_view":false,"can_edit":false,"can_edit_content":false,"is_private":false,"is_org_template_collection":false,"is_profile_template":false,"is_profile_collection":false,"pinned_and_locked":false,"is_submission_box_template":false,"is_submission_box_template_test":false,"is_inside_hidden_submission_box":false,"is_inside_a_submission":false,"is_subtemplate_or_instance":false,"show_language_selector":false,"template_num_instances":0,"awaiting_updates":null,"num_survey_responses":0,"max_row_index":null,"max_col_index":null,"frontend_url":"http://localhost/","common_viewable":null,"is_test_locked":null,"has_link_sharing":null,"is_restorable":false,"serializer":"SerializableCollection"},"relationships":{"roles":{"meta":{"included":false}},"collection_cover_items":{"meta":{"included":false}},"parent_collection_card":{"meta":{"included":false}},"parent":{"meta":{"included":false}},"live_test_collection":{"meta":{"included":false}},"submissions_collection":{"meta":{"included":false}},"submission_template":{"meta":{"included":false}},"collection_to_test":{"meta":{"included":false}},"organization":{"meta":{"included":false}},"created_by":{"meta":{"included":false}},"test_audiences":{"meta":{"included":false}},"test_results_collection":{"meta":{"included":false}},"collection_filters":{"meta":{"included":false}},"restorable_parent":{"meta":{"included":false}}}}}