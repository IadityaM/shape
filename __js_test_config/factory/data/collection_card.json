{"data":{"id":"2877260","type":"collection_cards","attributes":{"class_type":"CollectionCard::Primary","order":0,"width":1,"height":1,"parent_id":762904,"pinned":false,"image_contain":false,"card_question_type":null,"is_cover":false,"filter":"transparent_gray","hidden":false,"show_replace":true,"updated_at":"2019-12-16T23:14:34.478Z","col":1,"row":1,"section_type":null,"idea_id":null,"link":false,"pinned_and_locked":false,"can_edit_parent":false,"is_master_template_card":false},"relationships":{"item":{"meta":{"included":false}},"collection":{"meta":{"included":false}},"parent":{"meta":{"included":false}},"record":{"meta":{"included":false}},"templated_from":{"meta":{"included":false}}}}}