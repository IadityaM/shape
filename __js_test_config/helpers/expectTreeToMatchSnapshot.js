import toJson from 'enzyme-to-json'

const expectTreeToMatchSnapshot = tree => {
  expect(toJson(tree)).toMatchSnapshot()
}

export default expectTreeToMatchSnapshot
