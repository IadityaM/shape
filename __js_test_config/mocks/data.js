export const fakeTextItemAttrs = {
  id: 1,
  type: 'Item::TextItem',
  name: 'My Cool Item',
  text_data: 'This is the content for the item and it contains multiple sentences. Like this one.',
  breadcrumb: [['collections', 1, 'Some collection'], ['items', 1, 'my item']],
  parentPath: '/',
}
export const fakeTextItem = {
  ...fakeTextItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
  toJS: jest.fn().mockReturnValue(fakeTextItemAttrs),
}
export const fakeVideoItemAttrs = {
  id: 2,
  type: 'Item::VideoItem',
  name: 'Crazy Honey Badger',
  url: 'https://www.youtube.com/watch?v=4r7wHMg5Yjg'
}
export const fakeVideoItem = {
  ...fakeVideoItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeImageItemAttrs = {
  id: 3,
  type: 'Item::ImageItem',
  name: 'Earth from Space',
  filestack_file: {
    id: 1,
    url: 'https://www.nasa.gov/sites/default/files/styles/full_width_feature/public/thumbnails/image/iss052e023801_0.jpg',
  }
}
export const fakeImageItem = {
  ...fakeImageItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeItemCard = {
  id: 10,
  order: 0,
  height: 1,
  width: 1,
  record: fakeTextItem,
  item: fakeTextItem,
}
export const fakeCollectionCard = {
  id: 11,
  order: 1,
  height: 1,
  width: 1,
  record: fakeCollection,
  item: fakeCollection,
  reference: false,
}
export const fakeCollection = {
  id: 1,
  name: 'My Workspace X',
  type: 'Collection',
  breadcrumb: [],
  collection_cards: [
    fakeItemCard, fakeItemCard, fakeItemCard
  ]
}
export const fakePosition = {
  xPos: 0,
  yPos: 0,
  height: 1,
  width: 1,
}
