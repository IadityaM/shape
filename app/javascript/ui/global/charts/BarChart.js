import PropTypes from 'prop-types'
import { VictoryBar } from 'victory'

import ChartTooltip from '~/ui/global/charts/ChartTooltip'
import {
  datasetPropType,
  chartDomainForDatasetValues,
} from '~/ui/global/charts/ChartUtils'

const formatValues = (values, datasetOrder) => {
  return values.map((datum, i) => ({
    ...datum,
    x: datum.column,
    y: Math.max(datum.percentage, 0.5),
  }))
}

const BarChart = ({ dataset, cardArea, barsInGroup }) => {
  const values = formatValues(dataset.data, dataset.order)
  const { total, max_domain } = dataset
  const domain = chartDomainForDatasetValues({
    values,
    maxYDomain: max_domain,
  })
  const barWidth = barsInGroup > 3 ? 30 / (barsInGroup / 2) : 30
  // Only show labels if theres room for them
  const labelRenderer = datum =>
    barsInGroup > 4 ? () => {} : `${datum.percentage}%`
  const tooltipRenderer = datum => {
    let text = `${dataset.name}
    ${datum.percentage}%
    ${datum.value}/${total} total`
    if (total === 0) {
      text = 'No comparable chart\n from this feedback'
    }
    return text
  }
  return (
    <VictoryBar
      labels={d => d.percentage}
      labelComponent={
        <ChartTooltip
          tooltipTextRenderer={tooltipRenderer}
          labelTextRenderer={labelRenderer}
          cardArea={cardArea}
          displayTicks={false}
          alwaysShowLabels={true}
        />
      }
      padding={10 / barsInGroup}
      barWidth={barWidth}
      data={values}
      domain={domain}
      key={`dataset-${dataset.order}`}
      events={[
        {
          target: 'data',
          eventHandlers: {
            onMouseOver: () => [
              {
                target: 'labels',
                mutation: props => {
                  return { active: true }
                },
              },
            ],
            onMouseOut: () => [
              {
                target: 'labels',
                mutation: props => null,
              },
            ],
          },
        },
      ]}
    />
  )
}

BarChart.propTypes = {
  dataset: datasetPropType.isRequired,
  cardArea: PropTypes.number,
  barsInGroup: PropTypes.number,
}

BarChart.defaultProps = {
  cardArea: 1,
  barsInGroup: 1,
}

export default BarChart
