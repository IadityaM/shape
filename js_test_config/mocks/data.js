export const fakeItemAttrs = {
  id: 1,
  name: 'My Cool Item',
}
export const fakeItem = {
  ...fakeItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeCard = {
  id: 10,
  order: 0,
  height: 1,
  width: 1,
  record: fakeItem,
  item: fakeItem,
}
export const fakeCollection = {
  id: 1,
  name: 'My Workspace X',
  breadcrumb: [],
  collection_cards: [
    fakeCard, fakeCard, fakeCard
  ]
}
export const fakePosition = {
  xPos: 0,
  yPos: 0,
  height: 1,
  width: 1,
}
