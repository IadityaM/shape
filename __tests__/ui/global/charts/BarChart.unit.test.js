import BarChart from '~/ui/global/charts/BarChart'
import expectTreeToMatchSnapshot from '#/helpers/expectTreeToMatchSnapshot'
import { fakeAreaChartDataset } from '#/mocks/data'

const props = {}
let wrapper, render

describe('BarChart', () => {
  beforeEach(() => {
    props.dataset = fakeAreaChartDataset
    render = () => (wrapper = shallow(<BarChart {...props} />))
    render()
  })

  it('renders snapshot', () => {
    expectTreeToMatchSnapshot(wrapper)
  })

  it('renders VictoryBar', () => {
    expect(wrapper.find('VictoryBar').exists()).toBe(true)
  })
})
