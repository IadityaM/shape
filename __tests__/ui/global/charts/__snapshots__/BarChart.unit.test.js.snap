// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart renders snapshot 1`] = `
<VictoryBar
  barWidth={30}
  containerComponent={
    <VictoryContainer
      className="VictoryContainer"
      portalComponent={<Portal />}
      portalZIndex={99}
      responsive={true}
    />
  }
  data={
    Array [
      Object {
        "date": "2018-07-10",
        "label": undefined,
        "value": 10,
        "x": undefined,
        "y": NaN,
      },
      Object {
        "date": "2018-08-10",
        "label": undefined,
        "value": 25,
        "x": undefined,
        "y": NaN,
      },
      Object {
        "date": "2018-09-10",
        "label": undefined,
        "value": 30,
        "x": undefined,
        "y": NaN,
      },
    ]
  }
  dataComponent={
    <Bar
      defaultBarWidth={8}
      pathComponent={<Path />}
    />
  }
  domain={
    Object {
      "x": Array [
        1,
        3,
      ],
      "y": Array [
        0,
        30,
      ],
    }
  }
  events={
    Array [
      Object {
        "eventHandlers": Object {
          "onMouseOut": [Function],
          "onMouseOver": [Function],
        },
        "target": "data",
      },
    ]
  }
  groupComponent={
    <g
      role="presentation"
    />
  }
  key="dataset-0"
  labelComponent={
    <TickLabelWithTooltip
      alwaysShowLabels={true}
      cardArea={1}
      displayTicks={false}
      labelTextRenderer={[Function]}
      maxValue={0}
      minValue={0}
      tooltipTextRenderer={[Function]}
    />
  }
  labels={[Function]}
  padding={0}
  samples={50}
  scale="linear"
  sortOrder="ascending"
  standalone={true}
  theme={
    Object {
      "area": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#252525",
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "axis": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "axis": Object {
            "fill": "transparent",
            "stroke": "#252525",
            "strokeLinecap": "round",
            "strokeLinejoin": "round",
            "strokeWidth": 1,
          },
          "axisLabel": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 25,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
          "grid": Object {
            "fill": "none",
            "pointerEvents": "painted",
            "stroke": "none",
          },
          "tickLabels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "ticks": Object {
            "fill": "transparent",
            "size": 1,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "bar": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#252525",
            "padding": 8,
            "strokeWidth": 0,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "boxplot": Object {
        "boxWidth": 20,
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "max": Object {
            "padding": 8,
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "maxLabels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "median": Object {
            "padding": 8,
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "medianLabels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "min": Object {
            "padding": 8,
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "minLabels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "q1": Object {
            "fill": "#969696",
            "padding": 8,
          },
          "q1Labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "q3": Object {
            "fill": "#969696",
            "padding": 8,
          },
          "q3Labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "candlestick": Object {
        "candleColors": Object {
          "negative": "#252525",
          "positive": "#ffffff",
        },
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "chart": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "width": 450,
      },
      "errorbar": Object {
        "borderWidth": 8,
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "#252525",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "group": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "width": 450,
      },
      "legend": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "gutter": 10,
        "orientation": "vertical",
        "style": Object {
          "data": Object {
            "type": "circle",
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "title": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 5,
            "stroke": "transparent",
          },
        },
        "titleOrientation": "top",
      },
      "line": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "#252525",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "pie": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 400,
        "padding": 50,
        "style": Object {
          "data": Object {
            "padding": 10,
            "stroke": "transparent",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 20,
            "stroke": "transparent",
          },
        },
        "width": 400,
      },
      "scatter": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#252525",
            "stroke": "transparent",
            "strokeWidth": 0,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "stack": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "width": 450,
      },
      "tooltip": Object {
        "cornerRadius": 5,
        "flyoutStyle": Object {
          "fill": "#f0f0f0",
          "pointerEvents": "none",
          "stroke": "#252525",
          "strokeWidth": 1,
        },
        "pointerLength": 10,
        "style": Object {
          "fill": "#252525",
          "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
          "fontSize": 14,
          "letterSpacing": "normal",
          "padding": 5,
          "pointerEvents": "none",
          "stroke": "transparent",
          "textAnchor": "middle",
        },
      },
      "voronoi": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "transparent",
            "strokeWidth": 0,
          },
          "flyout": Object {
            "fill": "#f0f0f0",
            "pointerEvents": "none",
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 5,
            "pointerEvents": "none",
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
    }
  }
/>
`;
